app/views/registrations/show.html.erb

```diff
@@ -0,0 +1,274 @@
+<div class="grid-row">
+  <div class="column-full">
+    <%= render("waste_carriers_engine/shared/back", back_path: bo_path) %>
+
+    <div class="grid-row">
+      <div class="column-two-thirds">
+
+        <h1 class="heading-large">
+          <%= t(".heading", reg_identifier: @registration.reg_identifier) %>
+        </h1>
+
+        <% if @registration.pending? %>
+          <% if @registration.pending_manual_conviction_check? %>
+            <div class="panel">
+              <h2 class="heading-medium">
+                <%= t(".status.headings.pending_convictions") %>
+              </h2>
+              <p>
+                <%= t(".status.messages.pending_convictions_check") %>
+              </p>
+            </div>
+          <% elsif @registration.rejected_conviction_checks? %>
+            <div class="panel">
+              <h2 class="heading-medium">
+                <%= t(".status.headings.rejected") %>
+              </h2>
+              <p>
+                <%= t(".status.messages.rejected") %>
+              </p>
+            </div>
+          <% end %>
+
+          <% if @registration.pending_payment? %>
+            <div class="panel">
+              <h2 class="heading-medium">
+                <%= t(".status.headings.pending_payment") %>
+              </h2>
+              <p>
+                <%= t(".status.messages.pending_payment", total: display_pence_as_pounds(@registration.finance_details.balance)) %>
+              </p>
+              <%= link_to t(".status.actions.payment_button"), @registration.finance_details_link, class: 'button' %>
```

Comment:
Do we know why Hakiri (more likely brakeman under the covers) is seeing this line as

```ruby
link_to(t(".status.actions.payment_button"), RegistrationPresenter.new(WasteCarriersEngine::Registration.find_by(:reg_identifier => params[:reg_identifier]), view_context).finance_details_link, :class => "button")
```

I can see we get the registration in the controller using `params[:reg_identifier]` but after that everything we refer to in the presenter comes from the database. So I'd agree it's a "won't fix - what are you on about Hakiri!"

Unfortunately, I've just been in a classroom with [Ryan Ackroyd](https://en.wikipedia.org/wiki/Ryan_Ackroyd) for 2 days so now I'm a little paranoid about everything!

Reply:
I think Hakiri is complaining about the fact that the link will go to a different domain than the application but I am not super sure 🤔